File: 01_gpio_digital/led_blink_task.c
#include <wiringPi.h>
#define LED_PIN1 3
#define LED_PIN2 4
#define LED_PIN3 17
int main (void)
{
  //wiringPiSetup () ;
  wiringPiSetupGpio();
  pinMode (LED_PIN1, OUTPUT) ;
  pinMode (LED_PIN2, OUTPUT) ;
  pinMode (LED_PIN3, OUTPUT) ;
  for (int i=0; i<5; i++)
  {
    digitalWrite (LED_PIN1, HIGH) ; delay (500) ;
    digitalWrite (LED_PIN1,  LOW) ;
     digitalWrite (LED_PIN2, HIGH) ; delay (500) ;
    digitalWrite (LED_PIN2,  LOW) ;
    digitalWrite (LED_PIN3, HIGH) ; delay (500) ;
    digitalWrite (LED_PIN3,  LOW) ;
  }
  return 0 ;
}

File: 01_gpio_digital/led_blink_task.py
import RPi.GPIO as GPIO
import time

LED_PIN1 = 23
GPIO.setmode(GPIO.BCM)  #GPIO.BCM or GPIO.BOARD
GPIO.setup(LED_PIN1, GPIO.OUT)    #GPIO.OUT or GPIO.IN

LED_PIN2 = 24
GPIO.setup(LED_PIN2, GPIO.OUT)    #GPIO.OUT or GPIO.IN

LED_PIN3 = 25
GPIO.setup(LED_PIN3, GPIO.OUT)    #GPIO.OUT or GPIO.IN

for i in range(10):
    GPIO.output(LED_PIN1, GPIO.HIGH) # 1
    print("led on")
    time.sleep(2)
    GPIO.output(LED_PIN1, GPIO.LOW) # 0
    print("led off")
    time.sleep(0)
    GPIO.output(LED_PIN2, GPIO.HIGH) # 1
    print("led on")
    time.sleep(2)
    GPIO.output(LED_PIN2, GPIO.LOW) # 0
    print("led off")
    time.sleep(0)
    GPIO.output(LED_PIN3, GPIO.HIGH) # 1
    print("led on")
    time.sleep(2)
    GPIO.output(LED_PIN3, GPIO.LOW) # 0
    print("led off")
    time.sleep(0)

GPIO.cleanup()  # GPIO 핀상태 초기화    
















